paymentsheet/src/main/java/com/stripe/android/paymentsheet/PaymentOptionsItem.kt

```diff
@@ -10,32 +10,35 @@ internal sealed class PaymentOptionsItem {
 
     object AddCard : PaymentOptionsItem() {
         override val viewType: ViewType = ViewType.AddCard
-        override val isEnabledDuringEditing: Boolean = false
+        override val isEnabledDuringEditing: Boolean = true
     }
 
     object GooglePay : PaymentOptionsItem() {
         override val viewType: ViewType = ViewType.GooglePay
-        override val isEnabledDuringEditing: Boolean = false
+        override val isEnabledDuringEditing: Boolean = true
     }
 
     object Link : PaymentOptionsItem() {
         override val viewType: ViewType = ViewType.Link
-        override val isEnabledDuringEditing: Boolean = false
+        override val isEnabledDuringEditing: Boolean = true
     }
 
     /**
      * Represents a [PaymentMethod] that is already saved and attached to the current customer.
      */
     data class SavedPaymentMethod(
         val displayableSavedPaymentMethod: DisplayableSavedPaymentMethod,
+        private val canRemovePaymentMethods: Boolean,
     ) : PaymentOptionsItem() {
         override val viewType: ViewType = ViewType.SavedPaymentMethod
 
         val displayName = displayableSavedPaymentMethod.displayName
         val paymentMethod = displayableSavedPaymentMethod.paymentMethod
         val isModifiable: Boolean by lazy { displayableSavedPaymentMethod.isModifiable() }
 
-        override val isEnabledDuringEditing: Boolean = true
+        override val isEnabledDuringEditing: Boolean by lazy {
+            isModifiable || canRemovePaymentMethods
+        }
```

Comment:
Going to keep this but have it decided by the `isModifiable` and `canRemovePaymentMethods` boolean states for now. That way, it's at least much more unit testable from the `PaymentOptionsItemMapper` side.